config/vector.spec.toml

```diff
@@ -361,6 +361,45 @@ dns_servers = ["0.0.0.0:53"]
   oldest_first = false
   oldest_first = true
 
+# Ingests data through an internal data generator and outputs `log` events.
+[sources.generator]
+  # The component type. This is a required field that tells Vector which
+  # component to use. The value _must_ be `#{name}`.
+  #
+  # * required
+  # * type: string
+  # * must be: "generator"
+  type = "generator"
+
+  # The number of times to repeat outputting the `lines`.
+  #
+  # * optional
+  # * default: "infinite"
+  # * type: int
+  count = "infinite"
+
+  # The amount of time, in seconds, to pause between each batch of output lines.
+  # If not set, there will be no delay.
+  #
+  # * optional
+  # * no default
+  # * type: float
+  interval = 1.0
+
+  # The list of lines to output.
+  #
+  # * required
+  # * type: ["int"]
+  lines = ["Line 1", "Line 2"]
```

Comment:
This is not worth blocking the PR, but it would be nice to templatize this field so users should generate formatted test data. For example:

```
{{ timestamp }} - {{ host }} - {{ sequence }}
```

Libraries like [`faker`](https://github.com/cksac/fake-rs) could make this easy.

Reply:
Absolutely, that would be a great way of avoiding needing the sequence number setting.

Reply:
What do you think then? I think this would be a worthwhile upgrade.

Reply:
Do you want it for the initial implementation or as a follow up PR?

Reply:
Follow up is fine.